### DISCUSSION TITLE
23 gennaio 2023 B - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#27] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/abb6a520-60d1-40fa-b1bc-02ea4fbe295f)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/5082c10b-e2ad-4b42-b4b5-61c6fd94d7fb)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-14 10:47:50](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/05da4015-2e1b-4de5-b59c-7ec433ff5d41)
![IMAGE 2024-01-14 10:47:56](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/8e796e6b-fdc4-43e2-a32d-7a2c28fa15aa)

**Comment by atyion** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/101365532/4741833f-8516-4e2e-bbda-9f3e3bcc2d57)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/101365532/596be6d0-12cd-4225-b1b0-d4cff04192d0)

**Reply by IonutCicio** [reply]
Solo una pedanteria: secondo te è più corretto scrivere $D \nsubseteq ABC$ o $D \notin ABC$? E perché?

**Reply by atyion** [reply]
> Solo una pedanteria: secondo te è più corretto scrivere D⊈ABC o D∉ABC? E perché?

Ho sbagliato. Ho pensato che, in linea generale, una lettera potesse avere più attributi al suo interno (mai specificato), e che non era sbagliato neanche nel caso fosse semplicemente un attributo fare D⊈ABC. Nel primo caso è vero: basta guardare la defizione di sottoinsieme. Allo stesso tempo ho fatto un errore grossolano per quel che riguarda un attributo singolo, non ha letteralmente senso. 
Sinceramente non è una cosa alla quale ho posto un grandissimo pensiero sopra, ci sono andato a occhio.

**Reply by IonutCicio** [reply]
Andava bene anche come avevi fatto te, l'unica cosa che devi specificare _(mentalmente, non per forza sul foglio)_ è che in quel caso consideri $D$ come un singoletto (a.k.a $\set{ D }$, ma noi usiamo un sacco di abusi di notazione, quindi $D$ lo puoi vedere come un singoletto in base al contesto). Giusto perché non so quanto possa essere pedante il professore all'orale. 

P.S. le lettere tipo $A, B, C, D, ...$ le consideriamo come singoli attributi, le lettere tipo $X, Y, Z, V, W, S, ...$ le consideriamo come insiemi di attributi.

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/fa7c3a0f-e511-4484-bcb4-c7e0d4e3690c)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/9f1c3b66-2868-42ca-9e0e-3eb52e7ae0c7)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/dc6a0f0c-1d63-4099-9baf-c5354c884de4)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/42d97570-05e7-4d03-96cf-f6fd04e09270)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/155ce0d8-06d4-4419-98c6-916b12c40f43)

**Reply by salicius** [reply]
Ciao, non capisco il criterio con cui metti le b nella tabella, sulla A e la B sembrano corrette ma sulla C e la D come mai b1, non andrebbe b2 ?

**Comment by daromacarlo** [comment]
punto 2a)
![Immagine 2024-01-27 163606](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/cacb5dcc-60ed-45af-9146-740edf643047)
![Immagine 2024-01-27 163629](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/6e2e2717-1aed-4752-ac9b-d0f59c266517)
![Immagine 2024-01-27 163642](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/da7e964b-d253-441e-81ac-f2c438827e63)

**Reply by daromacarlo** [reply]
punto 2b)
![Immagine 2024-02-04 175318](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/f89aa5dc-0f22-4dd6-a3b7-486003f27712)
